sklearn/linear_model/cd_fast.pyx

```diff
@@ -201,26 +240,25 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 
                 if w_ii != 0.0:
                     # R += w_ii * X[:,ii]
-                    daxpy(n_samples, w_ii,
-                          <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                          1, <DOUBLE*>R.data, 1)
+                    axpy(n_samples, w_ii, &X_data[ii * n_samples], 1,
+                         R_data, 1)
 
                 # tmp = (X[:,ii]*R).sum()
-                tmp = ddot(n_samples,
-                           <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                           1, <DOUBLE*>R.data, 1)
+                if floating is float:
+                    tmp = dsdot(n_samples, &X_data[ii * n_samples], 1, R_data, 1)
```

Comment:
But if we use `dsdot` in [this line](https://github.com/scikit-learn/scikit-learn/pull/6913/files#diff-5d8b82d506a543bd684a1d403d5941e0R222), we may need to make both `R` and `y` become double, which will cause problems since `R` will be used to perform cblas operation with `X`.